provider/sgx.py

```diff
@@ -0,0 +1,189 @@
+        """
+        sgx_flags = self._params['sgx_flags'].split()
+        host_sgx_info = self._monitor.query_sgx_capabilities()
+        self.sgx_flags_verify(host_sgx_info, sgx_flags)
+
+        host_msg_epc_size = self._params["host_epc_size"]
+        host_sgx_section_size = host_sgx_info['section-size']
+        if host_msg_epc_size != int(host_sgx_section_size):
+            self._test.fail("Guest epc size %s is not equal to query_sgx"
+                            " section size %s" % (host_msg_epc_size,
+                                                  host_sgx_section_size))
+
+    def get_config_epc_size(self):
+        """
+        Get total epc size from configuration
+
+        :return: total configured epc size
+        """
+        guest_sgx_epc_list = self._params["vm_sgx_epc_devs"].split()
+
+        guest_config_epc_size = 0
+        for ele in guest_sgx_epc_list:
+            ele_memdev = self._params["vm_sgx_epc_memdev_%s" % ele]
+            tmp_epc_size = int(float(normalize_data_size(
+                self._params["size_mem_%s" % ele_memdev], "B", 1024)))
+            guest_config_epc_size += tmp_epc_size
+        return guest_config_epc_size
+
+    def guest_qmp_sgx_verify(self):
+        """
+        Verify guest sgx flags and epc size
+        """
+        sgx_flags = self._params['sgx_flags'].split()
+        guest_sgx_info = self._monitor.query_sgx()
+        self.sgx_flags_verify(guest_sgx_info, sgx_flags)
+        LOG_JOB.debug("Guest query SGX flags %s verified done", sgx_flags)
+
+        epc_sections_info = guest_sgx_info['sections']
+        guest_sgx_epc_list = self._params["vm_sgx_epc_devs"].split()
+
+        guest_config_epc_size = 0
+        for ele in guest_sgx_epc_list:
+            ele_memdev = self._params["vm_sgx_epc_memdev_%s" % ele]
+            tmp_epc_size = int(float(normalize_data_size(
+                self._params["size_mem_%s" % ele_memdev], "B", 1024)))
+            guest_config_epc_size += tmp_epc_size
+            epc_numa_id = int(self._params["vm_sgx_epc_node_%s" % ele])
+            tmp_epc_dict = {'size': tmp_epc_size, 'node': epc_numa_id}
+            if tmp_epc_dict in epc_sections_info:
+                epc_sections_info.remove(tmp_epc_dict)
+            else:
+                self._test.fail("%s is missing, qmp check failed." % str(
+                    tmp_epc_dict))
+        if len(epc_sections_info) != 0:
+            self._test.fail("The numbers of epc mis-match, qmp check failed.")
+
+        guest_sgx_section_size = guest_sgx_info['section-size']
+        if guest_config_epc_size != int(guest_sgx_section_size):
+            self._test.fail("Guest epc size %s is not equal to query_sgx"
+                            " section size %s" % (guest_config_epc_size,
+                                                  guest_sgx_section_size))
+        LOG_JOB.debug("Guest query SGX size verified successfully")
+
+    def guest_dmsg_sgx_verify(self):
```

Comment:
validate_epc_size

Reply:
guest_dmsg_sgx_verify verify guest sgx flags and epc size, so validate_epc_size is not describe all of functions